blackbox/docs/general/builtins/window-functions.rst

```diff
@@ -314,6 +314,56 @@ Example::
    +------+-----------------------------------------------+
    SELECT 4 rows in set (... sec)
 
+.. _window-function-lag:
+
+``lag(arg [, offset [, default] ])``
+------------------------------------
+
+.. note::
+
+   The ``lag`` window function is an :ref:`enterprise feature
+   <enterprise-features>`.
+
+Synopsis
+........
+
+::
+
+   lag(argument any [, offset integer [, default any]])
+
+Returns the argument value evaluated at the row that precedes the current row
+by the offset within the partition. If there is no such row, the return value
+is ``default``. If ``offset`` or ``default`` arguments are missing, they
+default to ``1`` and ``null``, respectively.
+
+Both ``offset`` and ``default`` are evaluated with respect to the current row.
+
+If ``offset`` is ``0``, then argument value is evaluated for the current row.
+
+The ``default`` and ``argument`` data types must match.
+
+Example::
+
+   cr> SELECT
+   ...   col1 as dept_id,
+   ...   col2 as year,
+   ...   col3 as budget,
+   ...   LAG(col3) OVER(PARTITION BY col1 ORDER BY col2) prev_budget
```

Comment:
Nice. 

Btw. for the future: Might also want to work with the new column aliases/relation alias feature. `unnest(..) as t (dept_id, year, budget)`. In a case like this it would have the advantage that the alias can already be used within the `LAG OVER(..)` declaration.